Service/DatabaseConfigurationService.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Becklyn\MysqlDumpBundle\Service;
+
+
+use Becklyn\MysqlDumpBundle\Entity\DatabaseConnection;
+use Doctrine\Bundle\DoctrineBundle\Registry;
+use Doctrine\DBAL\Connection;
+use Symfony\Component\DependencyInjection\ContainerAware;
+
+class DatabaseConfigurationService extends ContainerAware
+{
+    /**
+     * @var Registry
+     */
+    private $doctrine;
+
+
+    /**
+     * @var array
+     */
+    private $connectionIdentifiers;
+
+
+    /**
+     * @var string
+     */
+    private $backupPath;
+
+
+    /**
+     * ConfigurationService constructor.
+     *
+     * @param Registry $doctrine
+     * @param array    $connectionIdentifiers
+     * @param string   $backupPath
+     */
+    public function __construct (Registry $doctrine, array $connectionIdentifiers, $backupPath)
+    {
+        $this->doctrine              = $doctrine;
+        $this->connectionIdentifiers = $connectionIdentifiers;
+        $this->backupPath            = !empty($backupPath) ? $backupPath : '';
+    }
+
+
+    /**
+     * Returns all database connections that should be backed up by searching the configuration:
+     *  1) Uses, if provided, the arguments passed via CLI
+     *  2) Searches the config.yml for the key 'becklyn_mysql_dump:databases'
+     *  3) Searches the config.yml for the connections set up under 'doctrine:dbal:connections'
+     *
+     * @param array $cliArguments
+     *
+     * @return array An associative array of identifier => DatabaseConnection
+     */
+    public function getDatabases ($cliArguments)
+    {
+        // If any command line arguments are passed we prefer those ...
+        if (!empty($cliArguments))
+        {
+            return $this->getConnectionDataByIdentifier($cliArguments);
+        }
+
+        // ... alternatively fall back to the previously configured databases ...
+        if (!empty($databases = $this->getConfiguredDatabases()))
+        {
+            return $databases;
+        }
+
+        // ... or fallback to the full list of associated databases configured in Symfony.
+        return $this->getAppDatabases();
+    }
+
+
+    /**
+     * Returns a list of all databases that are configured under 'becklyn_mysql_dump:databases'
+     *
+     * @return array An associative array of identifier => DatabaseConnection
```

Comment:
You are right, we leave the map instead of a plain array.

But please update the typehint to identify the more concrete types:

The proper type would either be `(DatabaseConnection|null)[]` or `array<DatabaseConnection|null>`. You can opt for  `DatabaseConnection[]`, too, as the correct ones doesn't seem to be supported by PhpStorm yet. :(